Tests/SWBTaskConstructionTests/SwiftTaskConstructionTests.swift

```diff
@@ -3917,6 +3917,88 @@ fileprivate struct SwiftTaskConstructionTests: CoreBasedTests {
             results.checkNoDiagnostics()
         }
     }
+
+    @Test(.requireSDKs(.host))
+    func warningAsWarningsGroups() async throws {
+        try await withTemporaryDirectory { tmpDir in
+            let testProject = try await TestProject(
+                "ProjectName",
+                sourceRoot: tmpDir.join("srcroot"),
+                groupTree: TestGroup(
+                    "SomeFiles",
+                    children: [
+                        TestFile("File1.swift"),
+                    ]),
+                targets: [
+                    TestStandardTarget(
+                        "TargetName",
+                        type: .framework,
+                        buildConfigurations: [
+                            TestBuildConfiguration("Debug", buildSettings: [
+                                "SWIFT_WARNINGS_AS_WARNINGS_GROUPS": "-Wwarning",
+                                "SWIFT_EXEC": swiftCompilerPath.str,
+                                "CODE_SIGN_IDENTITY": ""
+                            ]),
+                        ],
+                        buildPhases: [
+                            TestSourcesBuildPhase([
+                                TestBuildFile("File1.swift"),
+                            ]),
+                        ])
+                ])
+
+            let tester = try await TaskConstructionTester(getCore(), testProject)
+
+            await tester.checkBuild(BuildParameters(configuration: "Debug", overrides: ["SWIFT_VERSION": "5.0"])) { results in
+                results.checkTarget("TargetName") { target in
+                    results.checkTask(.matchTarget(target), .matchRuleType("SwiftDriver Compilation")) { task in
+                        task.checkCommandLineContains(["-Wwarning"])
```

Comment:
Similarly, this check should assert that we see both the `-Wwarning` option and its argument on the resulting command line. So assuming the test used the `Unsafe` group, we'd check for `task.checkCommandLineContains(["-Wwarning", "Unsafe"])`